SpareMe/components/URLBar.js

```diff
@@ -53,57 +55,70 @@ class URLBar extends Component {
-                />
-                <TouchableOpacity
-                    style={styles.barButton}
-                    onPress={ () => {
-                        refreshHandler();
-                    }
-                }>
-                    <Image source={require('./refresh.png')} style={styles.refresh}/>
-                </TouchableOpacity>
-            </View>
-        );
+        const {backHandler, forwardHandler, refreshHandler, onChangeHandler, menuHandler, url} = this.props;
+          return (
+              <View style={styles.bar}>
+                  <TouchableOpacity
+                      disabled={!this.state.canGoBack}
+                      style={styles.barButton}
+                      onPress={ () => {
+                          backHandler();
+                      }
+                  }>
+                      <Image source={require('./back.png')} style={[styles.arrow, !this.state.canGoBack ? styles.disabled : null]}/>
+                  </TouchableOpacity>
+                  <TouchableOpacity
+                      disabled={!this.state.canGoForward}
+                      style={styles.barButton}
+                      onPress={ () => {
+                          forwardHandler();
+                      }
+                  }>
+                      <Image source={require('./forward.png')} style={[styles.arrow, !this.state.canGoForward ? styles.disabled : null]}/>
+                  </TouchableOpacity>
+                  <TextInput
+                      ref='textInput'
+                      style={styles.url}
+                      selectTextOnFocus={true}
+                      onChangeText={ (text) => {
+                          this.setState({
+                              url: text
+                          });
+                      }}
+                      onSubmitEditing={ () => {
+                          onChangeHandler(this.formattedURL());
+                      }}
+                      value={this.state.url}
+                      editable={true}
+                      autoCorrect={false}
+                      autoCapitalize='none'
+                      returnKeyType='go'
+                      underlineColorAndroid='transparent'
+                      keyboardType={Platform.OS === 'ios' ? 'web-search' : 'default'}
+                  />
+                  <TouchableOpacity
+                      style={styles.barButton}
+                      onPress={ () => {
+                          refreshHandler();
+                      }
+                  }>
+                      <Image source={require('./refresh.png')} style={styles.refresh}/>
+                  </TouchableOpacity>
+
+
+                  <Menu onSelect = {value => menuHandler(value)}>
+                    <MenuTrigger>
+                      <Image source={require('./menu.png')} style={styles.refresh}/>
+                    </MenuTrigger>
+                    <MenuOptions>
```

Comment:
These work well, but they're really difficult to tap. @wtravis1 can the menu options be made bigger and can the size of the menu be constrained to the length of the longest element?